src/check/parse/tokenize/src/main.zig

```diff
@@ -0,0 +1,1399 @@
+const std = @import("std");
+const GenCatData = @import("GenCatData");
+
+/// The token kinds
+pub const T = enum(u8) {
+    EndOfFile,
+
+    // primitives
+    Float,
+    String,
+    SingleQuote,
+
+    // a part of a string interpolation; generally you'll see something like:
+    // StringBegin, OpenCurly, <expr>, CloseCurly StringPart, OpenCurly, <expr>, CloseCurly
+    StringBegin,
+    StringPart,
+
+    // These are not technically valid, but we can have the formatter fix them up.
+    SingleQuoteBegin,
+    SingleQuotePart,
+
+    UpperIdent,
+    LowerIdent,
+    Underscore,
+    DotLowerIdent,
+    DotNumber,
+
+    OpenRound,
+    CloseRound,
+    OpenSquare,
+    CloseSquare,
+    OpenCurly,
+    CloseCurly,
+
+    OpPlus,
+    OpStar,
+    OpPizza,
+    OpAssign,
+    OpBinaryMinus, // trailing whitespace
+    OpUnaryMinus, // no trailing whitespace
+    OpNotEquals,
+    OpBang,
+    OpAnd,
+    OpAmpersand,
+    OpQuestion,
+    OpOr,
+    OpBar,
+    OpDoubleSlash,
+    OpSlash,
+    OpPercent,
+    OpCaret,
+    OpGreaterThanOrEq,
+    OpGreaterThan,
+    OpLessThanOrEq,
+    OpBackArrow,
+    OpLessThan,
+    OpEquals,
+    OpColonEqual,
+
+    Comma,
+    Dot,
+    DoubleDot,
+    TripleDot,
+    OpColon,
+    OpArrow,
+    OpBackslash,
+
+    // Keywords
+    KwIf,
+    KwThen,
+    KwElse,
+    KwWhen,
+    KwIs,
+    KwAs,
+    KwDbg,
+    KwCrash,
+    KwHas,
+    KwWhere,
+    KwImplements,
+    KwExposes,
+    KwImports,
+    KwImport,
+    KwWith,
+    KwGenerates,
+    KwPackage,
+    KwPackages,
+    KwRequires,
+    KwProvides,
+    KwTo,
+    KwInterface,
+    KwApp,
+    KwPlatform,
+    KwHosted,
+    KwDebug,
+    KwExpect,
+
+    NamedUnderscore,
+    OpaqueName,
+    Int,
+    DotUpperIdent,
+    NoSpaceDotNumber,
+    NoSpaceDotLowerIdent,
+    NoSpaceDotUpperIdent,
+};
+
+/// A helper function equivalent to the Rust impl T { is_keyword }.
```

Comment:
Let's not add comments that mention rust. They will just get stale super quick. Instead, let's try to add proper function comments.